tests/classes/form/validation/FormValidatorControlledVocabTest.php

```diff
@@ -3,86 +3,83 @@
 /**
  * @file tests/classes/form/validation/FormValidatorControlledVocabTest.php
  *
- * Copyright (c) 2014-2021 Simon Fraser University
- * Copyright (c) 2000-2021 John Willinsky
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
  * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
  *
  * @class FormValidatorControlledVocabTest
  *
- * @ingroup tests_classes_form_validation
- *
- * @see FormValidatorControlledVocab
- *
  * @brief Test class for FormValidatorControlledVocab.
  */
 
 namespace PKP\tests\classes\form\validation;
 
 use APP\core\Application;
-use PHPUnit\Framework\Attributes\CoversClass;
-use PHPUnit\Framework\MockObject\MockObject;
+use APP\facades\Repo;
+use Illuminate\Support\Facades\DB;
 use PKP\form\validation\FormValidatorControlledVocab;
-use PKP\controlledVocab\ControlledVocab;
-use PKP\controlledVocab\ControlledVocabDAO;
-use PKP\db\DAORegistry;
 use PKP\form\Form;
 use PKP\form\validation\FormValidator;
 use PKP\tests\PKPTestCase;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PKP\controlledVocab\ControlledVocab;
+use PKP\controlledVocab\ControlledVocabEntry;
 
 #[CoversClass(FormValidatorControlledVocab::class)]
 class FormValidatorControlledVocabTest extends PKPTestCase
 {
-    /**
-     * @see PKPTestCase::getMockedDAOs()
-     */
-    protected function getMockedDAOs(): array
-    {
-        return [...parent::getMockedDAOs(), 'ControlledVocabDAO'];
-    }
-
     public function testIsValid()
     {
         // Test form
         $form = new Form('some template');
 
-        // Mock a ControlledVocab object
-        /** @var ControlledVocab|MockObject */
-        $mockControlledVocab = $this->getMockBuilder(ControlledVocab::class)
-            ->onlyMethods(['enumerate'])
-            ->getMock();
-        $mockControlledVocab->setId(1);
-        $mockControlledVocab->setAssocType(Application::ASSOC_TYPE_CITATION);
-        $mockControlledVocab->setAssocId(333);
-        $mockControlledVocab->setSymbolic('testVocab');
+        $assocId = (DB::table("publications")
+            ->select("publication_id as id")
+            ->orderBy("publication_id", "desc")
+            ->first()
+            ->id ?? 0) + 100;
```

Comment:
rather than setting a fixed value like `333`, it's better to build one based on what we have in DB and add a const value (like here `100`) to just to be safe .